docs/TRACEZ_DESIGN.md

```diff
@@ -0,0 +1,175 @@
+# OpenTelemetry SDK Contrib - /tracez and /traceconfigz Design Doc
+
+This file contains information about the design choices for the OpenTelemetry /tracez and
+/traceconfigz zPages.
+
```

Comment:
Would this serve as the README.md page or is this something else? My personal take here is to keep this doc scoped to just a how-to guide and maybe a brief description of the different classes and what they do.

Since we are publishing an entry to Google's Open Source blog, I think a deep explanation of design decisions would be more appropriate there. We could even link to the blog entry from here once it's published.

Reply:
I believe this doc is specifically for design choices, how-to guide is the `README.md` file under `sdk_extensions/zpages/`

Reply:
Yeah this design doc drove the implementation of zpages so we want to keep it here in the community repo to refer to the motivation and to keep the history collected.